tools/validators/instance_validator/validate/entity_instance.py

```diff
@@ -143,32 +143,50 @@ def _ValidateTranslation(self):
         print('Invalid translation compliance string', translation_body.data)
         return False
 
+    # TODO(charbull): refactor this while refactoring the class
+    # validate fields
+    entity_type_str = str(self.entity['type'])
+    type_parse = entity_type_str.split('/')
+    namespace = type_parse[0]
+    entity_type = type_parse[1]
+    entity_type = self.universe.GetEntityType(namespace, entity_type)
+    all_fields_dict = entity_type.GetAllFields()
+
     # iterate through each translation device key and determine its form
-    for device_name in translation_body.keys():
+    for field_name in translation_body.keys():
       # check if keys are UDMI compliant
-      if isinstance(translation_body[device_name].data, str):
+      if isinstance(translation_body[field_name].data, str):
```

Comment:
Add comment why this case is skipped.  It's not immediately clear to my we wouldn't check every key.

Reply:
moved down, if it is udmi compliant not need to check the units and state